src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesLauncher.java

```diff
@@ -135,18 +135,6 @@ public void launch(SlaveComputer computer, TaskListener listener) {
                  Watch w2 = eventWatch(client, podName, namespace, runListener)) {
                 assert watcher != null; // assigned 3 lines above
                 watcher.await(template.getSlaveConnectTimeout(), TimeUnit.SECONDS);
-            } catch (InvalidDockerImageException e) {
-                Jenkins jenkins = Jenkins.get();
-                Queue q = jenkins.getQueue();
-                String runUrl = pod.getMetadata().getAnnotations().get("runUrl");
-                for (Queue.Item item: q.getItems()) {
-                    if (item.task.getUrl().equals(runUrl)) {
-                        q.cancel(item);
-                        break;
-                    }
-                }
-                slave.terminate();
-                return;
```

Comment:
Moved to https://github.com/jenkinsci/kubernetes-plugin/pull/772/files#diff-55ff61b56516ad2fa5dd36df80c3836cR243